src/heap_ptmalloc_2_27.cpp

```diff
@@ -1046,48 +1042,48 @@ read_malloc_state_by_symbol(address_t arena_vaddr, struct ca_malloc_state *mstat
 	/*
 	 * Extract data members of the arena metadata (struct malloc_state)
 	 */
-	if(!get_field_value(val, "flags", &data, false))
+	if(!ca_get_field_value(val, "flags", &data, false))
```

Comment:
with the std::optional, the code here needs to revise to something like:
```c++
value = ca_get_field_value(val, "flags")
if (!value)
   return false;
*data  = value.get()
```

Reply:
this will be a little longer than the original code but more clear.

Reply:
agree.